metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py

```diff
@@ -1024,6 +1024,18 @@ def _is_dialect_instance(
     return False
 
 
+def get_query_type(
```

Comment:
I think this method is not needed as it can be quite expensive to run query parser just to get the type of the query when you run the parser anyway in the init.
I think we should cache the temp table at the init when we parse all the query texts.

Reply:
As discussed over huddle lets try this at customer end and later we will improve it more